demo.py

```diff
@@ -25,115 +27,274 @@
+            - Nodes marked 'P': POI locations
+            - Blue nodes: new charger locations
+    """
+
+    fig, (ax1, ax2) = plt.subplots(1, 2)
+    fig.suptitle('New EV Charger Locations')
+    pos = {x: [x[0],x[1]] for x in G.nodes()}
+
+    # Locate POIs in map
+    poi_graph = G.subgraph(pois)
+    poi_labels = {x: 'P' for x in poi_graph.nodes()}
+
+    # Locate old charging stations in map
+    cs_graph = G.subgraph(charging_stations)
+
+    # Locate old charging stations at POIs in map
+    poi_cs_list = set(pois) - (set(pois)-set(charging_stations))
+    poi_cs_graph = G.subgraph(poi_cs_list)
+    poi_cs_labels = {x: 'P' for x in poi_graph.nodes()}
+    
+    # Draw old map (left image)
+    nx.draw_networkx(G, ax=ax1, pos=pos, with_labels=False, node_color='k', font_color='w')
+    nx.draw_networkx(poi_graph, ax=ax1, pos=pos, with_labels=True, 
+                        labels=poi_labels, node_color='k', font_color='w')
+    nx.draw_networkx(cs_graph, ax=ax1, pos=pos, with_labels=False, node_color='r',
+                        font_color='k')
+    nx.draw_networkx(poi_cs_graph, ax=ax1, pos=pos, with_labels=True, 
+                        labels=poi_cs_labels, node_color='r', font_color='w')
+
+    # Draw new map (right image)
+    new_cs_graph = G.subgraph(new_charging_nodes)
+    nx.draw_networkx(G, ax=ax2, pos=pos, with_labels=False, node_color='k', 
+                        font_color='w')
+    nx.draw_networkx(poi_graph, ax=ax2, pos=pos, with_labels=True, 
+                        labels=poi_labels,node_color='k', font_color='w')
+    nx.draw_networkx(cs_graph, ax=ax2, pos=pos, with_labels=False, node_color='r',
+                        font_color='k')
+    nx.draw_networkx(poi_cs_graph, ax=ax2, pos=pos, with_labels=True, 
+                        labels=poi_cs_labels,  node_color='r', font_color='w')
+    nx.draw_networkx(new_cs_graph, ax=ax2, pos=pos, with_labels=False, 
+                        node_color='#00b4d9', font_color='w')
+
+    # Save image
+    plt.savefig("map.png")
+
+if __name__ == '__main__':
+
+    # Collect user inputs
+    w, h, num_poi, num_cs, num_new_cs = read_in_args()
+
+    # Build large grid graph for city
+    G, pois, charging_stations, potential_new_cs_nodes = set_up_scenario(w, h, num_poi, num_cs)
+
+    # Build BQM
+    bqm = build_bqm(potential_new_cs_nodes, num_poi, pois, num_cs, charging_stations, num_new_cs)
+
+    # Run BQM on HSS
+    sampler = LeapHybridSampler()
+    print("\nRunning scenario on", sampler.solver.id, "solver...")
+    
+    new_charging_nodes = run_bqm_and_collection_solns(bqm, sampler, potential_new_cs_nodes)
+
+    # Compute stats on best solution found
+    poi_ave_dist, old_cs_ave_dist, new_cs_dist = compute_soln_stats(pois, num_poi, charging_stations, num_cs, new_charging_nodes, num_new_cs)
+
+    # Print results to commnand-line for user
+    printout_solution_to_cmdline(num_poi, new_charging_nodes, num_new_cs, num_cs, poi_ave_dist, old_cs_ave_dist, new_cs_dist)
```

Comment:
Since stats are used only for printing, why not compute them from there? That reduces a number of variables in flight (print fn already gets most of stats fn args).

Reply:
Because your initial feedback on the only PR was to have a separate function for computing stats... :)